server/kusk/client.go

```diff
@@ -64,3 +64,19 @@ func (k *kuskClient) GetApi(namespace, name string) (*kuskv1.API, error) {
 	}
 	return api, nil
 }
+
+// GetApiByFleet gets all APIs associated with the EnvoyFleet
+func (k *kuskClient) GetApiByEnvoyFleet(fleetNamespace, fleetName string) ([]kuskv1.API, error) {
+	list := kuskv1.APIList{}
+	if err := k.client.List(context.TODO(), &list, &client.ListOptions{}); err != nil {
+		return nil, err
+	}
+
+	toReturn := []kuskv1.API{}
+	for _, api := range list.Items {
```

Comment:
I tried doing this with field-selector option like:
```
# kubectl get api  --selector spec.fleet.name=default 
No resources found in default namespace.
```
even though such API instance exists:
```
# kubectl get api  -o yaml | grep "name: default" -B 2
  spec:
    fleet:
      name: default
```

I just couldn't get it to work. Maybe someone has an idea what I'm doing wrong.

Reply:
According to this [Stack Overflow post](https://stackoverflow.com/questions/55762084/how-can-i-find-the-list-of-field-selectors-supported-by-kubectl-for-a-given-reso) I think we need to implement our own function to fetch APIs by field selectors